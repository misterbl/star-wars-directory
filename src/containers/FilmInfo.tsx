import React, { PureComponent } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { getInfo } from "../selectors/appSelectors";
import { IAppState } from "../state";
import { IFIlmComponent } from "./FilmInfo.d";
import ROUTES from "../routes";
import formattedDate from "../utils/formattedDate";
import BackButton from "../components/BackButton";

export class FilmInfo extends PureComponent<IFIlmComponent> {
  backHome = () => {
    this.props.history.push(ROUTES.INDEX);
  };
  render() {
    if (this.props.film) {
      const {
        film: { release_date, director, opening_crawl, producer, title },
        history: { push }
      } = this.props;

      return (
        <>
          <BackButton
            route={ROUTES.INDEX}
            text="Back to search result"
            push={push}
          />
          <div className="d-flex m-5 p-5">
            <div className="avatar avatar__default film-image mr-3" />
            <div className="info-details p-3">
              <h2 className="text-center">{title.toUpperCase()}</h2>
              <p>{opening_crawl}</p>
              <p>{`Released the ${formattedDate(release_date)}`}</p>
              <p>{`Directed by: ${director}`}</p>
              <p>{`Produced by: ${producer}`}</p>
            </div>
          </div>
        </>
      );
    } else {
      return <>{this.backHome()}</>;
    }
  }
}
export const mapStateToProps = (state: IAppState) => ({
  film: getInfo(state)
});

export default withRouter(connect(mapStateToProps)(FilmInfo));
