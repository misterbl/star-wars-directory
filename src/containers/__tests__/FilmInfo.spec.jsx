/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import history from "../../testMocks/history.mock";
import { FilmInfo } from "../FilmInfo";
import ROUTES from "../../const/routes";

describe("FilmInfo", () => {
  const props = {
    history,
    film: {
      director: "director",
      opening_crawl: "1222",
      title: "title",
      release_date: "release_date",
      producer: "producer"
    }
  };
  const wrapper = shallow(<FilmInfo {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("redirect to index if there is no character", () => {
    wrapper.setProps({ film: null });
    expect(props.history.push).toHaveBeenCalledWith(ROUTES.INDEX);
  });
});
