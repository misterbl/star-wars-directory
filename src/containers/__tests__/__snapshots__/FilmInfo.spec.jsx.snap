// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmInfo matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmInfo
    film={
      Object {
        "director": "director",
        "opening_crawl": "1222",
        "producer": "producer",
        "release_date": "release_date",
        "title": "title",
      }
    }
    history={
      Object {
        "action": "PUSH",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 123,
        "listen": [MockFunction],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <BackButton
          push={[MockFunction]}
          route="/"
          text="Back to search result"
        />,
        <div
          className="d-flex m-5 p-5"
        >
          <div
            className="avatar avatar__default film-image mr-3"
          />
          <div
            className="info-details p-3"
          >
            <h2
              className="text-center"
            >
              TITLE
            </h2>
            <p>
              1222
            </p>
            <p>
              Released the NaNth undefined NaN
            </p>
            <p>
              Directed by: director
            </p>
            <p>
              Produced by: producer
            </p>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "push": [MockFunction],
          "route": "/",
          "text": "Back to search result",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="avatar avatar__default film-image mr-3"
            />,
            <div
              className="info-details p-3"
            >
              <h2
                className="text-center"
              >
                TITLE
              </h2>
              <p>
                1222
              </p>
              <p>
                Released the NaNth undefined NaN
              </p>
              <p>
                Directed by: director
              </p>
              <p>
                Produced by: producer
              </p>
            </div>,
          ],
          "className": "d-flex m-5 p-5",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "avatar avatar__default film-image mr-3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="text-center"
                >
                  TITLE
                </h2>,
                <p>
                  1222
                </p>,
                <p>
                  Released the NaNth undefined NaN
                </p>,
                <p>
                  Directed by: director
                </p>,
                <p>
                  Produced by: producer
                </p>,
              ],
              "className": "info-details p-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TITLE",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "TITLE",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1222",
                },
                "ref": null,
                "rendered": "1222",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Released the NaNth undefined NaN",
                },
                "ref": null,
                "rendered": "Released the NaNth undefined NaN",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Directed by: director",
                },
                "ref": null,
                "rendered": "Directed by: director",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Produced by: producer",
                },
                "ref": null,
                "rendered": "Produced by: producer",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BackButton
            push={[MockFunction]}
            route="/"
            text="Back to search result"
          />,
          <div
            className="d-flex m-5 p-5"
          >
            <div
              className="avatar avatar__default film-image mr-3"
            />
            <div
              className="info-details p-3"
            >
              <h2
                className="text-center"
              >
                TITLE
              </h2>
              <p>
                1222
              </p>
              <p>
                Released the NaNth undefined NaN
              </p>
              <p>
                Directed by: director
              </p>
              <p>
                Produced by: producer
              </p>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "push": [MockFunction],
            "route": "/",
            "text": "Back to search result",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="avatar avatar__default film-image mr-3"
              />,
              <div
                className="info-details p-3"
              >
                <h2
                  className="text-center"
                >
                  TITLE
                </h2>
                <p>
                  1222
                </p>
                <p>
                  Released the NaNth undefined NaN
                </p>
                <p>
                  Directed by: director
                </p>
                <p>
                  Produced by: producer
                </p>
              </div>,
            ],
            "className": "d-flex m-5 p-5",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "avatar avatar__default film-image mr-3",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="text-center"
                  >
                    TITLE
                  </h2>,
                  <p>
                    1222
                  </p>,
                  <p>
                    Released the NaNth undefined NaN
                  </p>,
                  <p>
                    Directed by: director
                  </p>,
                  <p>
                    Produced by: producer
                  </p>,
                ],
                "className": "info-details p-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TITLE",
                    "className": "text-center",
                  },
                  "ref": null,
                  "rendered": "TITLE",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1222",
                  },
                  "ref": null,
                  "rendered": "1222",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Released the NaNth undefined NaN",
                  },
                  "ref": null,
                  "rendered": "Released the NaNth undefined NaN",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Directed by: director",
                  },
                  "ref": null,
                  "rendered": "Directed by: director",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Produced by: producer",
                  },
                  "ref": null,
                  "rendered": "Produced by: producer",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
